operators/bosh-operator/DirectorService.js

```diff
@@ -836,7 +834,8 @@ class DirectorService extends BaseDirectorService {
         }
         logger.info(`[getCredentials] Fetching property from bosh for binding ${id}`);
         return this.getBindingProperty(deploymentName, id)
-          .then(binding => binding.credentials);
+          .then(binding => binding.credentials)
+          .tap(() => this.deleteBindingProperty(deploymentName, id));
```

Comment:
This is done to clear all old bindings from bosh? Any particular reason why it was moved here from `deleteBinding ` method?

Reply:
This was moved here to avoid Bosh call in normal flow. Also as we won't be creating new binding properties, deleteBindingProperty should not happen by default.

Reply:
I agree. Let us not have the deleteBindingProperty call at all.